targets/XPlatCppSdk/templates/PlayFab_Api.cpp.ejs

```diff
@@ -32,22 +41,29 @@ namespace PlayFab
 <%- getRequestActions("        ", apiCall) %>
         IPlayFabHttp& http = IPlayFabHttp::Get();
         const auto requestJson = request.ToJson();
-        http.AddRequest("<%- apiCall.url %>", <%- getAuthParams(apiCall) %>, requestJson, On<%- apiCall.name %>Result, SharedVoidPointer((callback == nullptr) ? nullptr : new ProcessApiCallback<<%- apiCall.result %>>(callback)), errorCallback, customData);
-    }
 
-    void PlayFab<%- api.name %>API::On<%- apiCall.name %>Result(CallRequestContainerBase& pRequest)
-    {
-        CallRequestContainer request = static_cast<CallRequestContainer&>(pRequest);
-        <%- apiCall.result %> outResult;
-        outResult.FromJson(request.errorWrapper.Data);
-        outResult.Request = request.errorWrapper.Request;
-<%- getResultActions("        ", apiCall) %>
-        const auto internalPtr = request.successCallback.get();
-        if (internalPtr != nullptr)
-        {
-            const auto callback = (*static_cast<ProcessApiCallback<<%- apiCall.result %>> *>(internalPtr));
-            callback(outResult, request.customData);
-        }
+        Json::FastWriter writer;
+        std::string jsonAsString = writer.write(requestJson);
+
+        std::unordered_map<std::string, std::string> headers;
+        headers.emplace(<%- getAuthParams(apiCall) %>);
+
+        http.AddPostRequest(
+            "<%- apiCall.url %>", 
+            headers,
```

Comment:
>headers [](start = 12, length = 7)

Be careful here passing in headers here that you created on the stack while in the API call. If you access those headers after that stack variable leaves the scope (and destroyed) there can be a problem